File: leetcode/rust/check-balanced-string.rs
impl Solution {
    pub fn is_balanced(num: String) -> bool {
        let mut even_sum = 0;
        let mut odd_sum = 0;

        for (i, c) in num.chars().enumerate() {
            let cur_num = c.to_digit(10).unwrap();

            match i % 2 == 0 {
                true => even_sum += cur_num,
                false => odd_sum += cur_num,
            };
        }

        even_sum == odd_sum
    }
}

File: leetcode/python/check-balanced-string.py
class Solution:
    def isBalanced(self, num: str) -> bool:
        even_sum = 0
        odd_sum = 0

        for i, c in enumerate(num):
            n = int(c)

            if i % 2 == 0:
                even_sum += n
            else:
                odd_sum += n

        return even_sum == odd_sum